docs/interface.rst

```diff
@@ -0,0 +1,106 @@
+Interface
+=========
+
+HTTP API
+--------
+
+Magic-Folder exposes all of its supported functionality in an HTTP API.
+The address of the HTTP server is part of the `daemon configuration`_.
+
+Authorization
+~~~~~~~~~~~~~
+
+The HTTP API is protected by Bearer token-style authorization scheme.
+Only requests which include the correct token will receive successful responses.
+Other requests will receive **401 Unauthorized** responses which omit the requested resource.
+The token value should be included with the **Bearer** scheme in the **Authorization** header element.
+For example::
+
+  Authorization: Bearer abcdefghijklmnopqrstuvwxyz
+
+The correct token value can be found in the *api_token* file inside the Magic-Folder daemon configuration directory.
+The token value is periodically rotated so clients must be prepared to receive an **Unauthorized** response even when supplying the token.
+In this case,
+the client should re-read the token from the filesystem to determine if the value held in memory has become stale.
+
+.. _`daemon configuration`: :ref:`config`
+
+``GET /v1/magic-folder``
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+This endpoint returns a list of all individual magic-folders managed by this daemon.
+
+The response code **OK** and the **Content-Type** is ``application/json``.
+The response body follows the form of this example::
+
+  { "folders":
+    [ { "name": "Alice's music", "local-path": "/home/alice/Music" }
+    , { "name": "Secret docs", "local-path": /home/alice/secrets" }
+    ]
+  }
+
+``GET /v1/snapshot``
+~~~~~~~~~~~~~~~~~~~~
+
+Retrieve information about all snapshots known to exist.
+
+The response is **OK** with an ``application/json`` **Content-Type**::
+
+  {"folder name": {
+      "foo/bar": [
+        { "type": "local"
+        , "identifier": "06be2d83-2d86-402d-ae2a-81b3779d72d9"
```

Comment:
Is this leading-comma syntax .. normal? (I've not seen it before).

Reply:
It's the usual syntax but I guess the formatting is uncommon.  It's motivated by two things:

  * Cleaner diffs since the comma and the element that requires it end up on the same line.  This means deleting a line doesn't require changing the line above it (and similar improvement for adding a line).
  * JSON doesn't allow a dangling comma (final comma with no element after it) so the pattern that works for, eg, Python import lists (with a comma after every element, even if there is no subsequent element) doesn't work here.